### PR TITLE
Bug fix:'operator-sdk add crd' doesn't work after add kubebuilder tags

### PR DESCRIPTION
**Pull request by xm2**
**Description of the change:**
Bug fix: 'operator-sdk add crd' doesn't work after add kubebuilder tags in _types.go

**Motivation for the change:**
1. In golang operator,  after add `kubebuilder` tags in _types.go files, e.g. validation tag or scale subresource tag, 
```diff
type AppServiceSpec struct {
	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	// +kubebuilder:validation:Maximum=100
+	// +kubebuilder:validation:Minimum=1
+	Replicas *int32 `json:"size,omitempty"`
}

// AppServiceStatus defines the observed state of AppService
// +k8s:openapi-gen=true
type AppServiceStatus struct {
	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	Replicas int32 `json:"replicas,omitempty"`
}

// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object

// AppService is the Schema for the appservices API
// +k8s:openapi-gen=true
// +kubebuilder:subresource:status
+// +kubebuilder:subresource:scale:specpath=.spec.size,statuspath=.status.replicas
type AppService struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   AppServiceSpec   `json:"spec,omitempty"`
	Status AppServiceStatus `json:"status,omitempty"`
}
```
2. run "operator-sdk generate k8s" 
3. delete "deploy/crds/app_v1alpha1_appservice_crd.yaml"
4. run `operator-sdk add crd  --api-version=app.example.com/v1alpha1 --kind=AppService` to re-generated crd yaml based on new kubebuilder tag, it failed:
```
$ rm deploy/crds/app_v1alpha1_appservice_crd.yaml
$ operator-sdk add crd  --api-version=app.example.com/v1alpha1 --kind=AppService
INFO[0000] Generating Custom Resource Definition (CRD) version app.example.com/v1alpha1 for kind AppService. 
Error: crd scaffold failed: (PROJECT file missing in dir /home/centos/go/src/app-operator)
Usage:
  operator-sdk add crd [flags]

Flags:
      --api-version string   Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)
  -h, --help                 help for crd
      --kind string          Kubernetes CustomResourceDefintion kind. (e.g AppService)

Global Flags:
      --verbose   Enable verbose logging
```

**Root cause analysis:**
   From investigation, this error is reported when the `Repo` in input.Config is not set and no "PROJECT" file in project root path when call `ValidateAndInitFields()` in `controller-tools/pkg/crd/generator/generator.go`.
   To fix the error, either add "Repo" in input.Config,  or,  add "PROJECT"file in project root path. 
   *BUT* "PROJECT" is a file in `kubebuilder` generated project, it doesn't exist in operator-sdk project.  

   So in the code change,  fix the bug by adding "Repo" in input.Config.

**Test Result**
After the fix, the _crd.yaml can be generated correctly with new "kubebuilder" tags:

```diff
$ rm deploy/crds/app_v1alpha1_appservice_crd.yaml
$ operator-sdk add crd  --api-version=app.example.com/v1alpha1 --kind=AppService
INFO[0000] Generating Custom Resource Definition (CRD) version app.example.com/v1alpha1 for kind AppService. 
INFO[0005] Created deploy/crds/app_v1alpha1_appservice_crd.yaml 
INFO[0005] RBAC rules in deploy/role.yaml already up to date for the resource (app.example.com/v1alpha1, AppService) 
INFO[0005] CRD generation complete.                     
$ git diff deploy/crds/app_v1alpha1_appservice_crd.yaml
diff --git a/deploy/crds/app_v1alpha1_appservice_crd.yaml b/deploy/crds/app_v1alpha1_appservice_crd.yaml
index 387e731..e01df42 100644
--- a/deploy/crds/app_v1alpha1_appservice_crd.yaml
+++ b/deploy/crds/app_v1alpha1_appservice_crd.yaml
@@ -11,6 +11,9 @@ spec:
     singular: appservice
   scope: Namespaced
   subresources:
+    scale:
+      specReplicasPath: .spec.size
+      statusReplicasPath: .status.replicas
     status: {}
   validation:
     openAPIV3Schema:
@@ -28,8 +31,26 @@ spec:
         metadata:
           type: object
         spec:
+          properties:
+            size:
+              description: 'INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+                Important: Run "operator-sdk generate k8s" to regenerate code after
+                modifying this file Add custom validation using kubebuilder tags:
+                https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html'
+              format: int32
+              maximum: 100
+              minimum: 1
+              type: integer
           type: object
         status:
+          properties:
+            replicas:
+              description: 'INSERT ADDITIONAL STATUS FIELD - define observed state
+                of cluster Important: Run "operator-sdk generate k8s" to regenerate
+                code after modifying this file Add custom validation using kubebuilder
+                tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html'
+              format: int32
+              type: integer
           type: object
   version: v1alpha1
   versions:
```

### CODE REVIEW

**Comment by LiliC** [comment]
cc @estroz ^

**Comment by AlexNPavel** [comment]
Does running `operator-sdk generate openapi` work? I believe that command should correctly regenerate the CRDs

**Comment by estroz** [comment]
@xm2 can you rebase onto/merge with `master`? We made a change in `master` that affects CI on older branches.

**Comment by hasbro17** [comment]
@xm2 Since this is a bug fix, can you please add a line to the Bugfix section of the CHANGELOG in the same format as the other changelogs.
https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#bug-fixes

@estroz PTAL.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ### Removed
 
 ### Bug Fixes
+- Fixes an issue that causes `operator-sdk add crd` to fail. ([#1660](https://github.com/operator-framework/operator-sdk/pull/1660))
```
Comment (estroz) [inline CHANGELOG.md]:
```suggestion
- Configure the repo path correctly in `operator-sdk add crd` and prevent it from being run outside of an operator project. ([#1660](https://github.com/operator-framework/operator-sdk/pull/1660))
```

Comment (estroz) [inline CHANGELOG.md]:
```suggestion
- Configure the repo path correctly in `operator-sdk add crd` and prevent the command from running outside of an operator project. ([#1660](https://github.com/operator-framework/operator-sdk/pull/1660))
```

**Comment by estroz** [comment]
@xm2 you'll have to merge your branch with master to fix the CI error unfortunately.

**Comment by xm2** [comment]
> @xm2 you'll have to merge your branch with master to fix the CI error unfortunately.

OK. @estroz, just synced my branch with master, all CI checks are OK now.

**Comment by estroz** [comment]
@xm2 awesome, and thanks for the PR!

**Pull request merged**